test/vote/vote.model.spec.js

```diff
@@ -0,0 +1,94 @@
+// vote.model.spec.js
+// Unit tests on the vote model
+"use strict";
+
+var mongoose = require('mongoose');
+
+
+var Vote = require(process.cwd() + '/src/vote/vote.model');
+
+var helperFile = require('../helper');
+let cleanDatabase = helperFile.cleanDatabase;
+
+
+var assert = require('chai').assert;
+
+
+// Test the casting of votes
+describe('Casting a size', function () {
+
+  beforeEach(function (done) {
+    cleanDatabase(function () {
+      done();
+    });
+  });
+
+  // We cast a vote of size 2
+  // At the end we expect to be given a vote object with the same
+  // details as we said
+  it('castVote, size 2 selected - not on a real story', function (done) {
+    // This should fail - vote must be dependant on an actual story!
+    var storyId = "dummy story id - not a mongo ID!!!"
+
+    Vote.create({
+      story: storyId,
+      size: 2
+    }, function (err, theVote) {
+      assert.isNotNull(err, "We should not accept bogus story ids")
+      done()
+    });
+
+  });
+
+
+  it('castVote, size 1 selected - on a real story', function (done) {
```

Comment:
Check what happens when you do not provide a size.
Check what happens when the story cannot be found
Check what happens when the story is is false ( your first if statement)